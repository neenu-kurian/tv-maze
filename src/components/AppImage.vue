<template>
  <div class="h-full">
    <img v-if="src" :src="src" :alt="alt" class="w-full h-full object-cover relative"/>
    <div v-else>
      <slot name="fallback">
        <span>{{ 'No Image Available' }}</span>
      </slot>
    </div>
  </div>
</template>

<script setup lang="ts">
import type { ImageProps } from '@/types/index.ts';

defineProps<ImageProps>();
</script>
