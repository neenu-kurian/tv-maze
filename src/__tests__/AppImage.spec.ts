import { render } from '@testing-library/vue';
import AppImage from '@/components/AppImage.vue';
import { describe, it, expect } from 'vitest';

describe('AppImage.vue', () => {
  it('matches snapshot', () => {
    const { container } = render(AppImage, {
      props: {
        src:'https://static.tvmaze.com/uploads/images/original_untouched/0/73.jpg' ,
        alt: 'Glee Image',
      },
    });
    expect(container).toMatchSnapshot();
  });
});
