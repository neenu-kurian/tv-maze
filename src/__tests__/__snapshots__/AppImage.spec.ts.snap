// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > matches snapshot 1`] = `
<div>
  <div
    class="h-full"
  >
    <img
      alt="Glee Image"
      class="w-full h-full object-cover relative"
      loading="lazy"
      src="https://static.tvmaze.com/uploads/images/original_untouched/0/73.jpg"
    />
  </div>
</div>
`;

exports[`AppImage.vue > matches snapshot 1`] = `
<div>
  <div
    class="h-full"
  >
    <img
      alt="Glee Image"
      class="w-full h-full object-cover relative"
      loading="lazy"
      src="https://static.tvmaze.com/uploads/images/original_untouched/0/73.jpg"
    />
  </div>
</div>
`;

exports[`HomeView.vue > matches snapshot 1`] = `
<div>
  <div
    class="h-full"
  >
    <img
      alt="Glee Image"
      class="w-full h-full object-cover relative"
      loading="lazy"
      src="https://static.tvmaze.com/uploads/images/original_untouched/0/73.jpg"
    />
  </div>
</div>
`;
